backend/conduit/articles/views.py

```diff
@@ -46,6 +46,24 @@ def get_articles(isPublished=None, tag=None, author=None, favorited=None, limit=
     return res.offset(offset).limit(limit).all()
 
 
+#Route to return articles authored by current user
+@blueprint.route('/api/profile/articles', methods=('GET',))
+@jwt_required
+@use_kwargs({'type': fields.Str()})
+@marshal_with(articles_schema)
+def get_user_articles(type=None):
+    res = Article.query
+    if type != "all":
+        if type == "drafts":
+            res = res.filter_by(isPublished=False)
+        elif type == "published":
+            res = res.filter_by(isPublished=True)
+        else:
+            raise InvalidUsage.article_not_found()
+    res = res.join(Article.author).join(User).filter(User.username == current_user.profile.user.username)
```

Comment:
You can shorten the query by just fetching the current user

Reply:
Done.